servers/assets/server.js

```diff
@@ -33,16 +40,32 @@ module.exports.run = () => {
     server.use('*', cors(corsOptions))
   }
 
+  // special middlewares
+  for (let middleware of additionalMiddlewares) {
+    middleware(server)
+  }
+
   // middlewares
   for (let key of Object.keys(middlewares)) {
     middlewares[key](server)
   }
 
-  server.listen(PORT, () => {
+  httpServer = server.listen(PORT, () => {
     console.info('server is running on http://localhost:' + PORT)
   })
+  return httpServer
+}
+
+const close = () => {
+  httpServer.close()
 }
 
-module.exports.close = () => {
-  server.close()
+module.exports = {
+  start,
+  close,
+  addMiddleware
 }
+
+process.on('SIGTERM', () => {
```

Comment:
Maybe log that process got `SIGTERM` signal. Could be useful for post mortem analysis.